Backend/ServerManagement.Tests/Azure/AzureServerManagerTests.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using ServerManagement.Azure;
+using ServerManagement.Azure.Configuration;
+using ServerManagement.Core.Exceptions;
+using ServerManagement.Core.Models;
+
+namespace ServerManagement.Tests.Azure;
+
+public class AzureServerManagerTests
+{
+    private readonly Mock<ILogger<AzureServerManager>> _mockLogger;
+    private readonly Mock<IOptions<AzureVmConfiguration>> _mockVmOptions;
+    private readonly Mock<IOptions<GameServerConfiguration>> _mockGameOptions;
+    private readonly Mock<HttpClient> _mockHttpClient;
+    private readonly AzureVmConfiguration _vmConfig;
+    private readonly GameServerConfiguration _gameConfig;
+
+    public AzureServerManagerTests()
+    {
+        _mockLogger = new Mock<ILogger<AzureServerManager>>();
+        _mockVmOptions = new Mock<IOptions<AzureVmConfiguration>>();
+        _mockGameOptions = new Mock<IOptions<GameServerConfiguration>>();
+        _mockHttpClient = new Mock<HttpClient>();
+
+        _vmConfig = new AzureVmConfiguration
+        {
+            SubscriptionId = "test-subscription-id",
+            ResourceGroupName = "test-rg",
+            VmName = "test-vm"
+        };
+
+        _gameConfig = new GameServerConfiguration
+        {
+            Host = "test-host",
+            Port = 26900,
+            TelnetPort = 8081,
+            AdminPassword = "test-password"
+        };
+
+        _mockVmOptions.Setup(x => x.Value).Returns(_vmConfig);
+        _mockGameOptions.Setup(x => x.Value).Returns(_gameConfig);
+    }
+
+    [Fact]
+    public void AzureServerManager_ShouldInitializeWithProperDependencies()
+    {
+        // Arrange & Act
+        var httpClient = new HttpClient();
+        var serverManager = new AzureServerManager(
+            _mockLogger.Object,
+            _mockVmOptions.Object,
+            _mockGameOptions.Object,
+            httpClient);
+
+        // Assert
```

Comment:
Comments like 'Assert' are not useful. Please remove any comment that doesn't add more value than the line of code it's documenting